runtime/ms_rest_azure/lib/ms_rest_azure/polling_state.rb

```diff
@@ -35,21 +35,35 @@ def initialize(azure_response, retry_timeout)
       update_response(azure_response.response)
       @resource = azure_response.body
 
+      case @response.status
```

Comment:
Agreed. Ran the cases with deleting a file from network and compute and confirmed it is successfully tested.